### PR TITLE
docs: add information about 'backend' flag

### LINKED ISSUE #3956: `--backend` flag is not mentioned in the docs
There is no mention of `--backend` flag in the docs https://docs.lfortran.org/en/usage/#invoking-lfortran. Reported here https://fortran-lang.discourse.group/t/fortran-on-web-using-lfortran/7957/3.

### PR DESCRIPTION
**Pull request by anushatomar13**
Fixed #3956

### CODE REVIEW

**Review by Pranavchiku — APPROVED** [review]
looks good to me, I'll wait for one more review before merging this

File: doc/src/usage.md [hunk 1]
```diff
@@ -74,6 +74,15 @@ compilation options, output options, link options and so on.
 * `--print-leading-space`, Print leading white space if format is unspecified
 * `--realloc-lhs`, Reallocate left hand side automatically
 * `--target <value>`, Generate code for the given target
+* `--backend` flag is used to specify the target backend for code generation in LFortran. The supported backends are:
+  
+  - `llvm`: The most advanced and default backend, used for generating LLVM IR or machine code through LLVM.
+  - `wasm`: A custom WebAssembly backend.
```
Comment (Shaikh-Ubaid) [inline doc/src/usage.md:80]: Let's mention: For generating `Webassembly` via our custom `wasm` backend.

**Review by Shaikh-Ubaid — APPROVED** [review]
Minor update shared. Rest looks good to me. Thanks for this!

**Comment by anushatomar13** [comment]
@Shaikh-Ubaid I have made the changes. Let me know if it's fine now.

File: doc/src/usage.md [hunk 2]
```diff
@@ -74,6 +74,15 @@ compilation options, output options, link options and so on.
 * `--print-leading-space`, Print leading white space if format is unspecified
 * `--realloc-lhs`, Reallocate left hand side automatically
 * `--target <value>`, Generate code for the given target
+* `--backend` flag is used to specify the target backend for code generation in LFortran. The supported backends are:
+  
+  - `llvm`: The most advanced and default backend, used for generating LLVM IR or machine code through LLVM.
+  - `wasm` : A custom backend for generating WebAssembly.
```
Comment (Shaikh-Ubaid) [inline doc/src/usage.md:80]: This seems fine. I think the wording I shared was better as it was in sync/similarity with wordings for the other backends below.

**Comment by anushatomar13** [comment]
@Shaikh-Ubaid Done, please check.

**Review by Shaikh-Ubaid — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**